internal/ci/vulnerability_result_diff.go

```diff
@@ -65,3 +65,32 @@ func DiffVulnerabilityResults(oldRes, newRes models.VulnerabilityResults) models
 
 	return result
 }
+
+// DiffVulnerabilityResultsByUniqueVulnCount will return a count of unique vulnerabilities that are in `newRes`
+// which is not present in `oldRes`, but not the reverse. This calculates the difference by vulnerability ID,
+// while ignoring the source of the vulnerability.
+func DiffVulnerabilityResultsByUniqueVulnCount(oldRes, newRes models.VulnerabilityResults) map[string]int {
+	oldResFlat := oldRes.Flatten()
+	newResFlat := newRes.Flatten()
+
+	oldResMap := map[string]int{}
+	newResMap := map[string]int{}
+
+	for _, vf := range oldResFlat {
+		oldResMap[vf.Vulnerability.ID] += 1
+	}
+
+	for _, vf := range newResFlat {
+		newResMap[vf.Vulnerability.ID] += 1
+	}
+
+	for k, oldVulnCount := range oldResMap {
+		// If the new result has less vulnerabilities than the old result remove the entry from the new result.
+		// `map`'s default value is 0 when empty, and delete also works fine when the entry is empty
+		if newResMap[k] <= oldVulnCount {
+			delete(newResMap, k)
+		}
+	}
+
+	return newResMap
```

Comment:
nit: should we just return len(newResMap) here? That's what the comment above the function also implies : that we're returning a count and not a map.

Reply:
Reworded it slightly, I still prefer to return the full map just to make testing a bit more accurate.